src/PolyType.Roslyn/IncrementalTypes/EquatableDiagnostic.cs

```diff
@@ -27,11 +27,16 @@ public readonly struct EquatableDiagnostic(
     /// </summary>
     public Location? Location { get; } = location?.GetLocationTrimmed();
 
+    /// <summary>
+    /// Additional locations for the diagnostic.
+    /// </summary>
+    public IEnumerable<Location>? AdditionalLocations { get; init; }
```

Comment:
Needs to be one of the special collection types to avoid reference equality:
```suggestion
    public ImmutableEquatableArray<Location>? AdditionalLocations { get; init; }
```

Reply:
`Location` does not satisfy the type constraint of `IEquatable<Location>`. But I'll update your Equals/GetHashCode methods here to make up for not using an equatable array type.